File: app/src/main/java/com/example/alcheringa2022/Merch.java
package com.example.alcheringa2022;

import android.content.Context;
import android.content.Intent;
import android.content.res.Resources;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.MotionEvent;
import android.view.View;
import android.view.ViewConfiguration;
import android.view.ViewGroup;
import android.widget.ProgressBar;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.alcheringa2022.Model.Merch_model;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.firestore.DocumentSnapshot;
import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.firestore.QuerySnapshot;

import java.util.ArrayList;
import java.util.List;
import java.util.zip.Inflater;

public class Merch extends Fragment implements onItemClick{
    RecyclerView recyclerView;
    List<Merch_model> merch_modelList;
    Merch_Adapter merch_adapter;
    FirebaseAuth firebaseAuth;
    FirebaseFirestore firestore;
    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        View view= inflater.inflate(R.layout.fragment_merch,container,false);
        recyclerView=view.findViewById(R.id.merch_recyclerview);
        firebaseAuth=FirebaseAuth.getInstance();
        firestore= FirebaseFirestore.getInstance();
        merch_modelList=new ArrayList<>();
        recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
        recyclerView.setHasFixedSize(true);
        merch_adapter=new Merch_Adapter(merch_modelList,getContext(),this);

        populate_merch();
        return  view;

    }

    private void populate_merch() {
        firestore.collection("Merch").get().addOnCompleteListener(new OnCompleteListener<QuerySnapshot>() {
            @Override
            public void onComplete(@NonNull Task<QuerySnapshot> task) {
                for(DocumentSnapshot documentSnapshot : task.getResult()){
                    merch_modelList.add(new Merch_model(documentSnapshot.getString("Name"),documentSnapshot.getString("Material")
                            ,documentSnapshot.getString("Price"),documentSnapshot.getString("Description"),"1",
                            documentSnapshot.getBoolean("Available"),documentSnapshot.getBoolean("Small"),
                            documentSnapshot.getBoolean("Medium"),documentSnapshot.getBoolean("Large"),
                            documentSnapshot.getBoolean("Large")));
                    recyclerView.setAdapter(merch_adapter);

                }
            }
        });


    }

    @Override
    public void Onclick(int position) {
        Toast.makeText(getContext(), "Ritesh", Toast.LENGTH_SHORT).show();
        Intent intent=new Intent(getActivity(),Merch_Description.class);
        intent.putExtra("item",merch_modelList.get(position));
        startActivity(intent);
    }

    @Override
    public void OnIncrementClick(int position) {

    }

    @Override
    public void OnDecrementClick(int position) {

    }
}

File: app/src/main/java/com/example/alcheringa2022/Merch.kt
package com.example.alcheringa2022

import android.os.Bundle
import com.example.alcheringa2022.Merch
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.Card
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import androidx.fragment.app.Fragment
import com.example.alcheringa2022.R
import com.example.alcheringa2022.databinding.FragmentMerchBinding
import com.example.alcheringa2022.ui.theme.Alcheringa2022Theme

class Merch : Fragment() {
    private var mParam1: String? = null
    private var mParam2: String? = null
    lateinit var binding: FragmentMerchBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (arguments != null) {
            mParam1 = requireArguments().getString(ARG_PARAM1)
            mParam2 = requireArguments().getString(ARG_PARAM2)
        }
    }
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        binding = FragmentMerchBinding.inflate(layoutInflater)
        return (binding.root)
    }
    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        binding.compose1.setContent {

        }
    }

    companion object {
        // the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
        private const val ARG_PARAM1 = "param1"
        private const val ARG_PARAM2 = "param2"
        fun newInstance(param1: String?, param2: String?): Merch {
            val fragment = Merch()
            val args = Bundle()
            args.putString(ARG_PARAM1, param1)
            args.putString(ARG_PARAM2, param2)
            fragment.arguments = args
            return fragment
        }
    }
    @Composable
    fun Merchandise(){
        Box(){
            Card(modifier = Modifier.fillMaxWidth(),
                    shape = RoundedCornerShape(8.dp),
                    elevation = 3.dp){
                Text(text = "VIPIN Jaluhtiria")
            }
            
        }
    }
}